holidays/countries/kazakhstan.py

```diff
@@ -31,48 +33,64 @@ def _populate(self, year):
         All holidays get observed on weekdays if they fall on weekends, but
         this has not been implemented as yet.
         """
-        # New Year's holiday (3 days)
+        # New Year's holiday (2 days)
         self[date(year, JAN, 1)] = "New Year"
         self[date(year, JAN, 2)] = "New Year Holiday"
-        self[date(year, JAN, 3)] = "New Year Holiday"
 
         # Orthodox Christmas
-        self[date(year, JAN, 7)] = "Orthodox Christmas"
+        if year >= 2007:
+            self[date(year, JAN, 7)] = "Orthodox Christmas"
 
         # Women's Day
         self[date(year, MAR, 8)] = "Women's Day"
 
         # Nauryz Holiday (3 days)
-        self[date(year, MAR, 21)] = "Nauryz"
-        self[date(year, MAR, 22)] = "Nauryz Holiday"
-        self[date(year, MAR, 23)] = "Nauryz Holiday"
+        if year >= 2009:
+            self[date(year, MAR, 21)] = "Nauryz"
+            self[date(year, MAR, 22)] = "Nauryz Holiday"
+            self[date(year, MAR, 23)] = "Nauryz Holiday"
 
-        # People Solidarity Holiday
-        self[date(year, MAY, 1)] = "People's Solidarity Day"
+        # Unity Day
+        self[date(year, MAY, 1)] = "Unity Day"
 
         # Defender's Day
-        self[date(year, MAY, 7)] = "Defender's Day"
+        if year >= 2013:
+            self[date(year, MAY, 7)] = "Defender's Day"
 
         # Victory Day
         self[date(year, MAY, 9)] = "Victory Day"
 
         # Capital City Day
-        self[date(year, JUL, 6)] = "Capital City Day"
-
-        # Kurban Ait
-        for hol_date in _islamic_to_gre(year, 12, 10):
-            self[hol_date] = "Kurban Ait"
+        if year >= 2009:
+            self[date(year, JUL, 6)] = "Capital City Day"
 
         # Constitution Day
-        self[date(year, AUG, 30)] = "Constitution Day"
+        if year >= 1996:
+            self[date(year, AUG, 30)] = "Constitution Day"
 
         # First President Day
-        self[date(year, DEC, 1)] = "First President Day"
+        if year >= 2013:
+            self[date(year, DEC, 1)] = "First President Day"
 
         # Independence Day (2 days)
         self[date(year, DEC, 16)] = "Independence Day"
         self[date(year, DEC, 17)] = "Independence Day Holiday"
 
+        if self.observed:
+            for k, v in list(self.items()):
+                if k.weekday() in WEEKEND and k.year == year:
+                    next_workday = k + rd(days=+1)
+                    while next_workday.weekday() in WEEKEND or self.get(
+                        next_workday
+                    ):
+                        next_workday += rd(days=+1)
+                    self[next_workday] = v + " (Observed)"
```

Comment:
Thess lines are not fully covered by test, although there is check with `self.holidays.observed = False`. @arkid15r, can you look into it? I tried new test functions :)

Reply:
Try
```
self.assertHoliday(Kazakhstan(observed=True), *observed_holidays)
self.assertNoHoliday(Kazakhstan(observed=False), *observed_holidays)
```

Reply:
It works. Thank you!